storage/range.go

```diff
@@ -1584,7 +1596,21 @@ func (r *Range) FirstIndex() (uint64, error) {
 
 // Snapshot implements the raft.Storage interface.
 func (r *Range) Snapshot() (raftpb.Snapshot, error) {
-	panic("TODO(bdarnell): implement raft.Storage.snapshot")
+	r.RLock()
+	defer r.RUnlock()
+
+	data, err := gogoproto.Marshal(r.Desc)
```

Comment:
Maybe add a todo here that we ought to be sending all of the data done the wire...? What's the plan here exactly?